dialects/math/math.thorin

```diff
@@ -0,0 +1,191 @@
+/// * `H`yperbolic counterpart
+/// * `A`rcus/`A`rea counterpart (inverse)
+///
+/// | Subtag | Alias        | A | H | R | FF | Meaning                   | Semantics               |
+/// |--------|--------------|---|---|---|----|---------------------------|-------------------------|
+/// | `ahff` |  `sin`       | o | o | o | oo | sine                      | \f$\sin x\f$            |
+/// | `ahfF` |  `cos`       | o | o | o | xo | cosine                    | \f$\cos x\f$            |
+/// | `ahFf` |  `tan`       | o | o | o | ox | tangent                   | \f$\tan x\f$            |
+/// | `ahFF` |              | o | o | o | xx | *unused*                  | -                       |
+/// | `aHff` |  `sinh`, `h` | o | x | o | oo | hyperbolic sine           | \f$\sinh x\f$           |
+/// | `aHfF` |  `cosh`      | o | x | o | xo | hyperbolic cosine         | \f$\cosh x\f$           |
+/// | `aHFf` |  `tanh`      | o | x | o | ox | hyperbolic tangent        | \f$\tanh x\f$           |
+/// | `aHFF` |              | o | x | o | xx | *unused*                  | -                       |
+/// | `Ahff` | `asin` , `a` | x | o | o | oo | arcus sine                | \f$\textrm{asin}\,x\f$  |
+/// | `AhfF` | `acos`       | x | o | o | xo | arcus cosine              | \f$\textrm{acos}\,x\f$  |
+/// | `AhFf` | `atan`       | x | o | o | ox | arcus tangent             | \f$\textrm{atan}\,x\f$  |
+/// | `AhFF` |              | x | o | o | xx | arcus *unused*            | -                       |
+/// | `AHff` | `asinh`      | x | x | o | oo | area hyperbolic sine      | \f$\textrm{asinh}\,x\f$ |
+/// | `AHfF` | `acosh`      | x | x | o | xo | area hyperbolic cosine    | \f$\textrm{acosh}\,x\f$ |
+/// | `AHFf` | `atanh`      | x | x | o | ox | area hyperbolic tangent   | \f$\textrm{atanh}\,x\f$ |
+/// | `AHFF` |              | x | x | o | xx | *unused*                  | -                       |
+///
+.ax %math.tri(ahff =  sin     , ahfF =  cos , ahFf =  tan , ahFF,
+              aHff =  sinh = h, aHfF =  cosh, aHFf =  tanh, aHFF,
+              Ahff = asin  = a, AhfF = acos , AhFf = atan , AhFF,
+              AHff = asinh    , AHfF = acosh, AHFf = atanh, AHFF):
+    Π [p: .Nat, e: .Nat] -> .Nat -> %math.F (p, e) -> %math.F (p, e), normalize_tri;
+///
+/// ### %math.pow
+///
+/// Power function: \f$x^y\f$
+///
+.ax %math.pow: Π [p: .Nat, e: .Nat] -> .Nat -> [%math.F (p, e), %math.F (p, e)] -> %math.F (p, e), normalize_pow;
+///
+/// ### %math.rt
+///
+/// [Square](https://en.wikipedia.org/wiki/Square_root) and [cube root](https://en.wikipedia.org/wiki/Cube_root):
+///
+/// | Name           | Meaning     | Semantics          |
+/// |----------------|-------------|--------------------|
+/// | `%%math.rt.sq` | square root | \f$\sqrt{x}\f$     |
+/// | `%%math.rt.cb` | cube root   |\f$\sqrt[3]{x}\f$   |
+///
+.ax %math.rt(sq,cb): Π [p: .Nat, e: .Nat] -> .Nat -> [%math.F (p, e), %math.F (p, e)] -> %math.F (p, e), normalize_rt;
```

Comment:
Later on (when we support definitions with namespaces in dialects), we should just define sqrt using pow to avoid duplication.